internal/rbd/types/mirror.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2024 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"context"
+
+	"github.com/ceph/ceph-csi/internal/util"
+
+	librbd "github.com/ceph/go-ceph/rbd"
+	"github.com/ceph/go-ceph/rbd/admin"
+)
+
+type Mirror interface {
```

Comment:
I called this as Mirror as its going to be common for both volume and group, let me know if this is fine or any other suggestion, I will keep this in Mind and add Destory and get in next implementation for now it looks like its not required and they are already part of volume and group which you are working on.

Reply:
Ok, `Mirror` is fine. We can always change it later it makes more sense then.

Yes, `Destroy()` is already part of the other types, so only required in case a function allocates/returns a `Mirror` type.